### ISSUE TITLE
max_compaction_bytes option does not take effect when manual compaction

### ISSUE THREAD

**Issue by jinshuaiyu2** [facebook/rocksdb#8053]
When I was looking at the manual compaction, I found that the max_compaction_bytes option would only limit the size of the input level files size and expand inputs，but it would not limit the size of the output_level_inputs size. Why?

**Comment by jay-zhuang** [comment]
Yes, for `CompactRange()` and `CompactFiles()`, we don't set `grandparents`:
https://github.com/facebook/rocksdb/blob/ec346da98c6326f65473331cec06818a6fa3d23b/db/compaction/compaction_picker.cc#L637
which is used to calculate the overlap to limit the future compaction size to `max_compaction_bytes`:
https://github.com/facebook/rocksdb/blob/27d57a035ec63ec401aac78c3be0bd9816fc0da4/db/compaction/compaction_job.cc#L218-L223
To limit the output file size, you can set the `CompactionOptions.output_file_size_limit`:
https://github.com/facebook/rocksdb/blob/d1c510baecc1aef758f91f786c4fbee3bc847a63/include/rocksdb/options.h#L1549

Closing the issue for now, you can also ask questions in our forum: https://groups.google.com/g/rocksdb thanks.

**Comment by ajkr** [comment]
> input level files size and expand inputs，but it would not limit the size of the output_level_inputs size

Yeah, I've noticed this before too. I was scared to fix it as it feels harmful for us to split compaction within a given level into several small compactions. Disadvantages include write-amp due to more files pulled into compaction due to partial overlap, and less opportunities for parallelism via subcompactions.
